core-bundle/src/OptIn/OptIn.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\OptIn;
+
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Model;
+use Contao\OptInModel;
+
+class OptIn implements OptInInterface
+{
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    public function __construct(ContaoFrameworkInterface $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $prefix, string $email, array $related): OptInTokenInterface
+    {
+        if (\strlen($prefix) > 6) {
```

Comment:
Why is that? DB field length?

Reply:
It's a custom prefix that is added at the beginning of the token so modules can tell their own tokens apart from others.

Reply:
The DB field is 24 characters long (hash) and the first six characters can be replaced with the prefix.